merkle/src/lib.rs

```diff
@@ -46,21 +74,68 @@ impl<F: Field, H: Hasher<F>> MerkleTree<F, H> {
     /// # Returns
     ///
     /// Returns an error type for easier "monadic style" chaining.
-    #[allow(dead_code)]
-    #[allow(clippy::result_unit_err)]
     pub fn verify_decommitment(
         &self,
         comm: H::Output,
         to_verify: &[(usize, H::Output)],
-        siblings: impl Iterator<Item = H::Output>,
-    ) -> Result<(), ()> {
-        if self.root_decommitment(to_verify, siblings).ok_or(())? == comm {
-            Ok(())
+    ) -> Result<bool, MerkleError> {
```

Comment:
This code assumes that you have all the nodes in the tree: the verifier does not.